docs/elasticsearch.txt

```diff
@@ -84,6 +84,73 @@ JanusGraph only uses default values for `index-name` and `health-request-timeout
 
 The REST client accepts the `index.[X].bulk-refresh` option. This option controls when changes are made visible to search. See https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-refresh.html[?refresh documentation] for more information.
 
+===== REST Client HTTPS support
+
+SSL support for HTTP can be enabled by setting the `index.[X].elasticsearch.ssl.enabled` configuration option to `true`. Note that depending on your configuration you may need to change the value of `index.[X].port` if your HTTPS port number is different from the default one for the REST API (9200).
+
+In addition to enabling SSL support, you may use non-default location for SSL Trust Store - for example if you run your own CA etc. This can be done as follows:
+
+[source, properties]
+----
+index.search.elasticsearch.ssl.truststore.location=/path/to/your/truststore.jks
+index.search.elasticsearch.ssl.truststore.password=truststorepwd
+----
+
+Note that these settings apply only to Elasticsearch REST client and do not affect any other SSL connections in JanusGraph.
+
+===== SSL client authentication for HTTPS
```

Comment:
I don't think this needs it's own section. If it were to be its own section I'd think it would be under the HTTPS section but the nesting is already pretty deep here.

Reply:
removed